### PR TITLE
Doc: Remove hard-coded edit_url links where possible

### PR DESCRIPTION
**Pull request by karenzone**
Improvements in Elastic.co docs tooling (approximately 2019) made the "Edit" links better at guessing the source URL for editing. Hard-coded links for Logstash core documentation are no longer required, and can potentially cause problems (as in the recent `master`->`main` branch renaming). Taking this opportunity to do more cleanup. 

Note that links to source files for plugin docs (in `logstash-docs`) must remain hard-coded because they live in their own repos. 

**PREVIEW:** https://logstash_13636.docs-preview.app.elstc.co/guide/en/logstash/master/introduction.html

Related: #13619

### CODE REVIEW

**Review by dedemorton — APPROVED** [review]
LGTM. I spot checked the pages (did not review all of them), but I think the approach is sound.

One thing I remember, though, is having to add empty lines between include statements because I ran into a weird bug many years ago. Sometimes the doc build was unable to resolve attributes used in section IDs. I never figured out what was causing the problem, but it was incredibly difficult to debug. It's possible (likely) the coding bug has been fixed, but given the complexity of debugging build problems, I'd preserve the extra lines unless you know the bug is fixed.

File: docs/index.asciidoc [hunk 1]
```diff
@@ -225,75 +135,46 @@ include::{plugins-repo-dir}/plugins/filters.asciidoc[]
...
-
-:edit_url!: 
+// Submitting a Plugin 
 include::static/submitting-a-plugin.asciidoc[] 
-
-:edit_url!:
-include::static/listing-a-plugin.asciidoc[]
-
-:edit_url!: 
+include::static/listing-a-plugin.asciidoc[]
```
Comment (dedemorton) [inline docs/index.asciidoc:159]: One other thing...did you mean to include this file, too?

**Comment by karenzone** [comment]
I made a couple of assumptions that were possibly risky...
1. All the work/changes for the asciidoctor conversion might have fixed any problems caused by removing empty lines.
2. If problems were still there, they would be obvious and easy to attribute to removing empty lines. 

As we never identified the bug(s), I can't 100% guarantee that asciidoctor and other doc system improvements fixed them.  If troubleshooting is as tricky as you say, perhaps I'll leave them in for now and take the win with the broader body of work here. Thanks!

File: docs/index.asciidoc (hunk 1 above)
Comment (karenzone) [reply docs/index.asciidoc:159]: Yes. Thank you for checking. :-)

**Pull request merged**